ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -629,8 +631,12 @@ def _get_next_disk_id(self):
 
     # ---------------------------------------------------------------------
     def _launch_helper_instance(self):
-        """Launch the helper instance that is used to create the new image"""
         self._set_zone_to_use()
+        """When using an already running instance, simply look up this one"""
```

Comment:
This logic should not be here. the method has a purpose, tat purpose is to launch the helper instance and thus it is named accordingly..

2 options:
Put the logic around where _launch_helper_instance is called i.e.
if self.running_ID:
    helper_instance = self._get_instance(self.running_ID)
else:
    helper_instance = self._launch_helper_instance()

or implement a generic implementation

def _get_helper_instance(self):
     if self.running_id:
         return ......
     else:
         return self._launch_helper_instance()

The current implementation introduces a method that has multiple purposes, look up or creation.